iot-hub/Samples/service/RoleBasedAuthenticationSample/Program.cs

```diff
@@ -0,0 +1,48 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+using Azure.Identity;
+using CommandLine;
+using Microsoft.Azure.Devices;
+using System;
+using System.Threading.Tasks;
+
+namespace RoleBasedAuthenticationSample
+{
+    public class Program
+    {
+        /// <summary>
+        /// A sample to illustrate how to use Azure active directory for authentication to the IoT hub.
+        /// <param name="args">Run with `--help` to see a list of required and optional parameters.</param>
+        public static async Task Main(string[] args)
+        {
+            // Parse application parameters
+            Parameters parameters = null;
+            ParserResult<Parameters> result = Parser.Default.ParseArguments<Parameters>(args)
+                .WithParsed(parsedParams =>
+                {
+                    parameters = parsedParams;
+                })
+                .WithNotParsed(errors =>
+                {
+                    Environment.Exit(1);
+                });
+
+            // Initialize Azure active directory credentials.
+            Console.WriteLine("Creating token credential.");
+
+            // These environment variables are necessary for DefaultAzureCredential to use application Id and client secret to login.
+            Environment.SetEnvironmentVariable("AZURE_CLIENT_SECRET", parameters.ClientSecret);
+            Environment.SetEnvironmentVariable("AZURE_CLIENT_ID", parameters.ClientId);
+            Environment.SetEnvironmentVariable("AZURE_TENANT_ID", parameters.TenantId);
+
+            // DefaultAzureCredential supports different authentication mechanisms and determines the appropriate credential type based of the environment it is executing in.
+            // It attempts to use multiple credential types in an order until it finds a working credential.
+            // For more info <see cref="https://docs.microsoft.com/en-us/dotnet/api/azure.identity?view=azure-dotnet"/>
+            var tokenCredential = new DefaultAzureCredential(); 
```

Comment:
I'd consider making the ```var``` into ```TokenCredential``` so that it is more obvious that users don't have to use ```DefaultAzureCredential``` when constructing their service client

Reply:
What? I don't get this.

Reply:
We don't do that in track2. This is how it was suggested by the Azure SDK team.

Reply:
At most, I'd put a link to all the different kinds of TokenCredential types and options.